client/post-editor/post-editor.jsx

```diff
@@ -951,6 +969,10 @@ export const PostEditor = React.createClass( {
 
 		if ( mode === 'html' ) {
 			this.editor.setEditorContent( content );
+
+			if ( this.state.selectedText ) {
+				this.getSelectedText();
```

Comment:
It took me a minute to figure out why `getSelectedText()` was being called when switching to HTML view. A comment here to note that it's being reset would be helpful.